src/productdetails.ts

```diff
@@ -20,24 +28,111 @@ function productDetails(): void {
   function renderStep(data: DataProduct[]): void {
     shop.loadProduct(data)
 
-    const idProduct = getProductIdQuery()
+    const productID = getProductIDQuery()
+
+    if (isValidID(productID)) {
+
+      renderProduct(productID)
+
+      // Listener para el boton "Buy now"
+      addListenerAddCart()
+
+      // Listener para el icono de Carrito, abre o cierra la lista.
+      addListenerDisplayCart()
 
-    renderProductDetails(idProduct)
-    renderCommentsList(idProduct)
+      // Listener para Abrir la lista de menu cuando toca el anchor del mensaje al añadir algo al carrito.
+      addListenerSeeCart()
+
+      renderCommentsList(productID)
+      return
+    }
+    // Si no se encuentra el id se redirige a la pagina notfound
+    window.location.replace('notfound.html')
+    return
   }
 
-  function getProductIdQuery(): string {
+  function getProductIDQuery(): string {
     let searchQuery: string = window.location.search;
     const searchTerm: string = '='
-    const idProduct: string = searchQuery.slice(searchQuery.indexOf(searchTerm)).replace(searchTerm, '')
-    return idProduct
+    const productID: string = searchQuery.slice(searchQuery.indexOf(searchTerm)).replace(searchTerm, '')
+    return productID
+  }
+
+  function isValidID(productID: string): boolean {
```

Comment:
Me choco un poco el nombre, yo lo daria a esta funcion un nombre como `checkValidId`. xD

Reply:
Si la función da un booleano era una buena manera de nombrar una función empezando con `is[...]` o eso había visto en un video en ingles. Como que al leerlo la función te dice `esValidoElID?()` o algo así.